src/wazuh_db/wdb_agent.c

```diff
@@ -664,140 +576,129 @@ int wdb_update_agent_multi_group(int id, char *group) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        wdb_close_global();
-        return -1;
-    }
+    int result = 0;
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
+    int wdb_sock = -1;
 
-    sqlite3_bind_text(stmt, 1, name, -1, NULL);
+    snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_INSERT_AGENT_GROUP], name);
+    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
 
-    if (wdb_step(stmt) == SQLITE_DONE)
-        result = (int)sqlite3_last_insert_rowid(wdb_global);
+    if (result == 0)
+        result = wdb_find_group(name);
     else {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
+        mdebug1("SQLite query error: %s", wdbquery);
         result = -1;
     }
 
-    sqlite3_finalize(stmt);
     return result;
 }
 
-/* Update agent belongs table. It opens and closes the DB. Returns number of affected rows or -1 on error. */
+/* Update agent belongs table. It opens and closes the DB. Returns 1 or -1 on error. */
 int wdb_update_agent_belongs(int id_group, int id_agent) {
     int result = 0;
-    sqlite3_stmt *stmt;
-
-    if (wdb_open_global() < 0)
-        return -1;
-
-    if (wdb_prepare(wdb_global, SQL_INSERT_AGENT_BELONG, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        return -1;
-    }
-
-    sqlite3_bind_int(stmt, 1, id_group);
-    sqlite3_bind_int(stmt, 2, id_agent);
-
-    result = wdb_step(stmt) == SQLITE_DONE ? (int)sqlite3_changes(wdb_global) : -1;
-    sqlite3_finalize(stmt);
-
-    return result;
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
+    int wdb_sock = -1;
+
+    snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_INSERT_AGENT_BELONG], id_group, id_agent);
+    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
+  
+    return result == 0 ? 1 : -1;
 }
 
-/* Delete agent belongs table. It opens and closes the DB. Returns number of affected rows or -1 on error. */
+/* Delete agent belongs table. It opens and closes the DB. Returns 1 or -1 on error. */
 int wdb_delete_agent_belongs(int id_agent) {
     int result = 0;
-    sqlite3_stmt *stmt;
-
-    if (wdb_open_global() < 0)
-        return -1;
-
-    if (wdb_prepare(wdb_global, SQL_DELETE_AGENT_BELONG, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        return -1;
-    }
-
-    sqlite3_bind_int(stmt, 1, id_agent);
-
-    result = wdb_step(stmt) == SQLITE_DONE ? (int)sqlite3_changes(wdb_global) : -1;
-    sqlite3_finalize(stmt);
-
-    return result;
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
+    int wdb_sock = -1;
+
+    snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_DELETE_AGENT_BELONG], id_agent);
+    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
+ 
+    return result == 0 ? 1 : -1;
```

Comment:
Log information about the result and make use of `OS_SUCCESS` and `OS_INVALID`

Reply:
Done